zipkin4net-aspnetcore/Criteo.Profiling.Tracing.Middleware/TracingHandler.cs

```diff
@@ -27,10 +27,10 @@ internal TracingHandler(ITraceInjector<HttpHeaders> injector, string serviceName
             {
                 if (clientTrace.Trace != null)
                 {
-                    _injector.Inject(clientTrace.Trace, request.Headers);
+                    _injector.Inject(clientTrace.Trace.Child(), request.Headers);
```

Comment:
I think this needs a test as it isn't exactly clear what this does. Is this making a child of the client trace? Usually client traces are leaves in the tree...